### PR TITLE
Processing data PR

### PR DESCRIPTION
**Pull request by Kossi-stack**
**In this pull request these are the following requirement**

Created model classes for all entities as shown in the ERD diagram. 
    Methods that are implemented on models :

  -  Users:
    A method that returns the 3 most recent posts for a given user. 

   - Posts:
    A method that updates the posts counter for a user. 
    A method that returns the 5 most recent comments for a given post. 

   - Comments:
    A method that updates the comments counter for a post. 

   - Likes:
    A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments, foreign_key: 'post_id'
+  has_many :likes, foreign_key: 'post_id'
+
+  def update_post_counter
+    user.increment!(:posts_counter)
+  end
```
Comment (DeliceLydia) [inline app/models/post.rb]: ## Optional
- [x] I suggest that you use one of the Rails Active Callbacks `after_save` to call the `update_post_counter` method every time a new post has been saved to the database, then make the `update_post_counter` method private

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post, foreign_key: 'post'
+
+  def update_comments_counter
+    post.increment!(:comments_counter)
+  end
```
Comment (DeliceLydia) [inline app/models/comment.rb:9]: ## Optional
- [x] I suggest that you use one of the Rails Active Callbacks `after_save` to call the `update_comments_counter` method every time a new post has been saved to the database, then make the `update_comments_counter` method private

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post, foreign_key: 'post_id'
+
+  def update_likes_counter
+    post.increment!(:likes_counter)
+  end
```
Comment (DeliceLydia) [inline app/models/like.rb:9]: ## Optional
- [x] I suggest that you use one of the Rails Active Callbacks `after_save` to call the `update_likes_counter` method every time a new post has been saved to the database, then make the `update_likes_counter` method private

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @Kossi-stack, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
- [x] Please implement `current_user` method in the `Application controller`


### Cheers and Happy coding!👏👏👏

Feel free to leave a question, or a comment on the PR thread if something is not clear 100%. 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lameck1 — CHANGES_REQUESTED** [review]
Hi, @Kossi-stack, 

Well done working on the optional changes suggested in the previous review 👏🏼.
You have however not worked on the single required change that was mentioned in the summary under `Required Changes ♻️` as the one we have below. Please look back at the previous comment.

 ## Required Changes ♻️
- As mentioned in the previous review, please fulfill the following project requirements.

> Additionally, you will need to make current user data available to all controllers.
> 
> - Create a method called current_user in your ApplicationController.
> - A current_user method should return the first user from a database.
> - Don't worry about proper authentication at this time. You will add it in the upcoming projects.

## Optional suggestions

- NA


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lameck1 — APPROVED** [review]
Hi, @Kossi-stack,

Well done on working on the previous review concerns and suggestions 👏🏼.
The tasks for this milestone are now successfully implemented 💯.
All the best with the next milestone.

You are free to merge this branch :shipit:

Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**